### ISSUE TITLE
Media picker only show preview of svg on load

### ISSUE THREAD

**Issue by bjarnef** [umbraco/Umbraco-CMS#6414]
When selecting a svg image, it only seems to show the preview of this on load, but not previewing the image after selection or when changing from e.g. a txt file to svg file.

![2019-09-24_10-45-56](https://user-images.githubusercontent.com/2919859/65496526-eba02300-deb8-11e9-8b16-0556bba7755b.gif)

**Comment by bjarnef** [comment]
The json of the media element when the preview does work on load.

```
{
  "name": "Cherry Svgrepo Com",
  "id": 1147,
  "udi": "umb://media/3cc19dbd4d7d414684abae25357d047f",
  "icon": "icon-document",
  "trashed": false,
  "key": "3cc19dbd-4d7d-4146-84ab-ae25357d047f",
  "parentId": -1,
  "alias": null,
  "path": "-1,1147",
  "metaData": {
    "ContentTypeAlias": "File",
    "MediaPath": "/media/uwoncpd3/cherry-svgrepo-com.svg",
    "IsContainer": false
  },
  "extension": "svg",
  "thumbnail": "/media/uwoncpd3/cherry-svgrepo-com.svg"
}
```

The json after selecting the svg from media picker. After save and refresh the json above is shown.

```
{
  "name": "Cherry Svgrepo Com",
  "id": 1147,
  "udi": "umb://media/3cc19dbd4d7d414684abae25357d047f",
  "icon": "icon-document",
  "trashed": false,
  "key": "3cc19dbd-4d7d-4146-84ab-ae25357d047f",
  "parentId": -1,
  "alias": null,
  "path": "-1,1147",
  "metaData": {
    "ContentTypeAlias": "File",
    "MediaPath": "/media/uwoncpd3/cherry-svgrepo-com.svg",
    "IsContainer": false
  },
  "thumbnail": null,
  "image": "/media/uwoncpd3/cherry-svgrepo-com.svg",
  "isFolder": false,
  "extension": "svg",
  "width": "150",
  "height": 150,
  "aspectRatio": 1,
  "selectable": true,
  "flexStyle": {
    "flex": "1 1 150px",
    "max-width": "150px",
    "min-width": "100px",
    "min-height": "100px"
  },
  "selected": true
}
```

**Comment by bjarnef** [comment]
Selection of a  `txt` file from media picker returns this json:

```
{
  "name": "Breadcrumbs",
  "id": 1148,
  "udi": "umb://media/959bbc289acf499097f6441c67617f81",
  "icon": "icon-document",
  "trashed": false,
  "key": "959bbc28-9acf-4990-97f6-441c67617f81",
  "parentId": -1,
  "alias": null,
  "path": "-1,1148",
  "metaData": {
    "ContentTypeAlias": "File",
    "MediaPath": "/media/gc4jxfop/breadcrumbs.txt",
    "IsContainer": false
  },
  "thumbnail": null,
  "image": "/media/gc4jxfop/breadcrumbs.txt",
  "isFolder": false,
  "extension": "txt",
  "width": "150",
  "height": 150,
  "aspectRatio": 1,
  "selectable": true,
  "flexStyle": {
    "flex": "1 1 150px",
    "max-width": "150px",
    "min-width": "100px",
    "min-height": "100px"
  },
  "selected": true
}
```

After save and refresh:

```
{
  "name": "Breadcrumbs",
  "id": 1148,
  "udi": "umb://media/959bbc289acf499097f6441c67617f81",
  "icon": "icon-document",
  "trashed": false,
  "key": "959bbc28-9acf-4990-97f6-441c67617f81",
  "parentId": -1,
  "alias": null,
  "path": "-1,1148",
  "metaData": {
    "ContentTypeAlias": "File",
    "MediaPath": "/media/gc4jxfop/breadcrumbs.txt",
    "IsContainer": false
  },
  "extension": "txt",
  "thumbnail": null
}
```

**Comment by bjarnef** [comment]
```
$scope.clickHandler = function (image, event, index) {
     console.log("image", image);
     ...
});

![image](https://user-images.githubusercontent.com/2919859/65498243-d7a9f080-debb-11e9-8c7a-0933ee87a606.png)

We should probably resolve the thumbnail for svg here as well as we do in media picker property editor.
https://github.com/umbraco/Umbraco-CMS/blob/v8/dev/src/Umbraco.Web.UI.Client/src/views/propertyeditors/mediapicker/mediapicker.controller.js#L81-L83

Furthermore I would like to change the naming of funcition and variables using `image` to `media` since the selected media is not necessary an image.
https://github.com/umbraco/Umbraco-CMS/blob/v8/dev/src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/mediapicker/mediapicker.controller.js

.. and `image.isFolder` sounds strange 😄 
https://github.com/umbraco/Umbraco-CMS/blob/v8/dev/src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/mediapicker/mediapicker.controller.js#L194

**Comment by nul800sebastiaan** [comment]
Aaand the question you knew was coming: is this v8.1.5 or some other version?

**Comment by bjarnef** [comment]
@nul800sebastiaan it is with latest changes in `v8/dev` branch :)

**Comment by bjarnef** [comment]
Okay, regading the property `image` in json for other filetypes, it is a bit confusing, because it could be other types of media like folder and file like `txt`.
```"image": "/media/gc4jxfop/breadcrumbs.txt"```

**Comment by bjarnef** [comment]
@nul800sebastiaan I have submitted a PR to fix this issue and a few other issues I noticed in mediapicker property editor and infinite editor https://github.com/umbraco/Umbraco-CMS/pull/6418

**Comment by nul800sebastiaan** [comment]
Fixed in https://github.com/umbraco/Umbraco-CMS/pull/6418

**Comment by nul800sebastiaan** [comment]
Uses just the required fixed for 8.2.1 in https://github.com/umbraco/Umbraco-CMS/commit/df6e29a24931f3214cdbd50015c6dd4ce2b73695
